admin/client/ColorSchemeDropdown.tsx

```diff
@@ -0,0 +1,129 @@
+import React = require("react")
+import { computed, action } from "mobx"
+import Select, { ValueType } from "react-select"
+import { ColorScheme, ColorSchemes } from "charts/ColorSchemes"
+import { observer } from "mobx-react"
+
+export interface ColorSchemeOption {
+    colorScheme?: ColorScheme
+    gradient?: string
+    label: string
+    value: string
+}
+
+export interface ColorSchemeDropdownProps {
+    additionalColorSchemes: ColorSchemeOption[]
+    defaultValue?: string
+    gradientColorCount: number
+    invertedColorScheme: boolean
+    onChange: (selected: ColorSchemeOption) => void
+}
+
+@observer
+export class ColorSchemeDropdown extends React.Component<
+    ColorSchemeDropdownProps
+> {
+    static defaultProps = {
+        additionalColorSchemes: [],
+        gradientColorCount: 6
+    }
+
+    @computed get availableColorSchemes() {
+        const { additionalColorSchemes, gradientColorCount } = this.props
+
+        return additionalColorSchemes.concat(
+            Object.entries(ColorSchemes).map(([key, scheme]) => {
+                return {
+                    colorScheme: scheme as ColorScheme,
+                    gradient: this.createLinearGradient(
+                        scheme as ColorScheme,
+                        gradientColorCount
+                    ),
+                    label: (scheme as ColorScheme).name,
+                    value: key
+                }
+            })
+        )
+    }
+
+    createLinearGradient(colorScheme: ColorScheme, count: number) {
+        const colors = colorScheme.getColors(count)
+
+        const step = 100 / count
+        const gradientEntries = colors.map(
+            (color, i) => `${color} ${i * step}%, ${color} ${(i + 1) * step}%`
+        )
+
+        return `linear-gradient(90deg, ${gradientEntries.join(", ")})`
+    }
+
+    @action.bound onChange(selected: ValueType<ColorSchemeOption>) {
+        // The onChange method can return an array of values (when multiple
+        // items can be selected) or a single value. Since we are certain that
+        // we are not using the multi-option select we can force the type to be
+        // a single value.
+
+        this.props.onChange(selected as ColorSchemeOption)
+    }
+
+    @action.bound formatOptionLabel(option: ColorSchemeOption) {
```

Comment:
I'm not quite sure if `action.bound` is the right thing to do here, because this is not really an action ;)
The alternative is probably to assign `this.formatOptionLabel.bind(this)` below - which one do you like better?

Reply:
There is a `@bind` decorator that we use from the "decko" package. You can import it like this:

```js
import { bind } from "decko"
```

That's what you're trying to achieve right?

Reply:
Ah, great, thank you!